zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -2,22 +2,36 @@
 //!
 //! [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
 
+use num_integer::div_ceil;
+
 use std::convert::TryFrom;
+use std::str::FromStr;
 
 use zebra_chain::{
     amount::{Amount, Error, NonNegative},
     block::Height,
-    parameters::Network,
+    parameters::{Network, NetworkUpgrade::*},
+    transaction::Transaction,
+    transparent::{address::ToAddressWithNetwork, Address, Output},
 };
 
 use crate::block::subsidy::general::{block_subsidy, halving_divisor};
-use crate::parameters::subsidy::FOUNDERS_FRACTION_DIVISOR;
+use crate::parameters::subsidy::*;
+
+/// Returns `true` if we are in the founders reward period of the blockchain.
+pub fn founders_reward_active(height: Height, network: Network) -> bool {
+    let canopy_activation_height = Canopy
+        .activation_height(network)
+        .expect("Canopy activation height is known");
+
+    height < canopy_activation_height && halving_divisor(height, network) == 1
```

Comment:
We've changed this code twice now, so let's add a comment for future reference.

```suggestion
    // The Zcash Specification and ZIPs explain the end of the founders reward in different ways,
    // because some were written before the set of Canopy network upgrade ZIPs  was decided.
    // These are the canonical checks recommended by `zcashd` developers.
    height < canopy_activation_height && halving_divisor(height, network) == 1
```